### PR TITLE
Split Create User

### PR DESCRIPTION
**Pull request by idafurjes**
**What this PR does / why we need it**:
This PR is splitting the sqlstore method CreateUser into a service and a store. By splitting the method org and userorg service/store was added. I decided to not to remove the CreateUser method for testing purposes.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/user/userimpl/userimpl.go
Comment (papagian) [inline pkg/services/user/userimpl/userimpl.go:91]: I think that this should be inside a transaction so that if it fails to insert user to organisation then the user inserted also is rollbacked.

File: pkg/services/user/userimpl/userimpl.go
Comment (papagian) [inline pkg/services/user/userimpl/userimpl.go:41]: I think that if the user exists then this it should fail: that's what happens in the existing implementation:
https://github.com/grafana/grafana/blob/32d4f6ac60f3169ddb3128d55145cad644fa7a3e/pkg/services/sqlstore/user.go#L53-L55

File: pkg/services/orguser/orguserimpl/store.go
Comment (spinillos) [inline pkg/services/orguser/orguserimpl/store.go:31]: I think that you can omit the error check and return `orgID, err` directly.

File: pkg/services/user/userimpl/store.go
Comment (spinillos) [inline pkg/services/user/userimpl/store.go:61]:
```suggestion
	return usr, err
```

File: pkg/services/org/orgimpl/store.go
Comment (spinillos) [inline pkg/services/org/orgimpl/store.go:66]:
```suggestion
	return orgID, err
```

File: pkg/services/orguser/model.go
Comment (mildwonkey) [inline pkg/services/orguser/model.go:14]: Question f(or my other PR: do we need swagger directives in the models generally, or is this a special case?

**Review by mildwonkey — APPROVED** [review]
LGTM, once the merge conflicts are resolved. 

I'm *definitely* going to wait until you merge this to do anything else with my PR (but don't rush); too many overlapping files changed

**Review by zserge — APPROVED** [review]
Not a small one 😩 . But acceptable, most of the changes are automatic due to the renames, and new code for org/user looks good ✅

**Pull request merged**
